### PR TITLE
v0.6.0: update apikey logic [BAC-788]

### PR DESCRIPTION
**Pull request by samweinberg23**
Not sure how best to add coverage without fully emulating the integration tests...Definitely integration tests will cover this but still :/

### CODE REVIEW

File: src/dydx-client.ts [hunk 1]
```diff
@@ -69,15 +75,17 @@ export class DydxClient {
...
+          host: this.host,
+          apiKey: this.apiKey,
+          secret: this.secret,
+          passphrase: this.passphrase,
+          starkPrivateKey: this.starkPrivateKey,
+        });
       } else {
         return notSupported(
-          'Private endpoints are not supported since apiPrivateKey was not provided',
+          'Private endpoints are not supported since apiKey, secret or passphrase were not provided',
```
Comment (Kenadia) [inline src/dydx-client.ts:88]: `were` -> `was`

File: src/dydx-client.ts [hunk 2]
```diff
@@ -13,22 +13,26 @@ import {
...
   readonly host: string;
   readonly apiTimeout?: number;
-  readonly apiPrivateKey?: string | KeyPair;
+  readonly apiKey?: string;
   readonly ethSendOptions?: EthereumSendOptions;
   readonly networkId: number;
   readonly starkPrivateKey?: string | KeyPair;
   readonly web3?: Web3;
+  readonly secret?: string;
+  readonly passphrase?: string;
```
Comment (Kenadia) [inline src/dydx-client.ts:35]: Our client options are fairly complex, I think it's worth organizing these as a group:

```
export interface ApiKeyCredentials {
  key: string,
  secret: string,
  passphrase: string,
}
...
apiKey?: ApiKeyCredentials
```

File: src/modules/private.ts [hunk 3]
```diff
@@ -47,16 +50,28 @@ const METHOD_ENUM_MAP: Record<RequestMethod, ApiMethod> = {
...
+    secret: string,
+    passphrase: string,
     starkPrivateKey?: string | KeyPair,
-  ) {
+  }) {
     this.host = host;
-    this.apiKeyPair = asSimpleKeyPair(asEcKeyPair(apiPrivateKey));
+    this.apiKey = apiKey;
+    this.secret = secret;
+    this.passphrase = passphrase;
```
Comment (Kenadia) [inline src/modules/private.ts:74]: (same as above, I think we can group these as an object)

File: src/modules/private.ts [hunk 4]
```diff
@@ -515,6 +531,17 @@ export default class Private {
...
   }
 
+  /**
+   * @description get the apiKeys associated with an ethereumAddress
+   *
+   * @param ethereumAddress the apiKeys are for
+   */
+  async getApiKeys(
+    ethereumAddress: string,
+  ): Promise<{ apiKeys: ApiKeyResponseObject[] }> {
```
Comment (Kenadia) [inline src/modules/private.ts:541]: this is a different kind of response object (compare `NewApiKey` vs `ApiKey` in our api spec)

File: __tests__/api-keys.test.ts [hunk 5]
```diff
@@ -10,13 +10,13 @@ import { asMock } from './helpers/util';
 
 describe('API Keys Module', () => {
 
-  it('signs a request', async () => {
+  it.skip('signs a request', async () => {
```
Comment (Kenadia) [inline __tests__/api-keys.test.ts:13]: (unskip before merge)

**Review by Kenadia — COMMENTED** [review]
There are some things we can unit test, e.g.:
1. create client with API key creds
2. call a private function
3. check that axios was called with the expected signature and other headers

See `api-keys.test.ts`, let's just make a similar `private.test.ts` that tests one GET endpoint and one POST endpoint

Note that axios is already mocked by `jest.setup.js`

File: __tests__/crypto-signing.test.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+    const client = new DydxClient('https://example.com', { web3, apiKeyCredentials });
+    await client.private.getApiKeys(
+      account.address,
+    );
+    expect(client.private.sign({
+      requestPath: '/v3/api-keys?ethereumAddress=0xE5714924C8C5c732F92A439075C8211eB0611aaC',
+      method: RequestMethod.GET,
+      isoTimestamp: '2021-02-01T19:38:54.508Z',
+    })).toEqual('jGElyQttdQDNqlRu5CpCtfVEYcikknzXWsOjKJAcTtI=');
```
Comment (samweinberg23) [inline __tests__/crypto-signing.test.ts:34]: I tested this against librarian and this string was actually generated in librarian (same with the secret)

File: src/types.ts [hunk 7]
```diff
@@ -138,6 +146,8 @@ export interface ApiFastWithdrawal extends ApiStarkwareSigned {
 
 export interface ApiKeyResponseObject {
   apiKey: string;
```
Comment (Kenadia) [inline src/types.ts:148]: Did we change this on our backend to `key`, so `ApiKeyResponseObject` and `ApiKeyCredentials` are now the same?

**Review by Kenadia — APPROVED** [review]

File: src/types.ts (hunk 7 above)
Comment (samweinberg23) [reply src/types.ts:148]: We did - let me update this

**Pull request closed without merging**